src/LinkContainer.js

```diff
@@ -1,61 +1,45 @@
-// This is largely taken from react-router/lib/Link.
-
-import React from 'react';
-
-function isLeftClickEvent(event) {
-  return event.button === 0;
-}
-
-function isModifiedEvent(event) {
-  return !!(
-    event.metaKey ||
-    event.altKey ||
-    event.ctrlKey ||
-    event.shiftKey
-  );
-}
-
-function createLocationDescriptor(to, query, hash, state) {
-  if (query || hash || state) {
-    return { pathname: to, query, hash, state };
-  }
-
-  return to;
-}
-
-const propTypes = {
-  onlyActiveOnIndex: React.PropTypes.bool.isRequired,
-  to: React.PropTypes.oneOfType([
-    React.PropTypes.string,
-    React.PropTypes.object,
-  ]).isRequired,
-  query: React.PropTypes.string,
-  hash: React.PropTypes.string,
-  state: React.PropTypes.object,
-  action: React.PropTypes.oneOf([
-    'push',
-    'replace',
-  ]).isRequired,
-  onClick: React.PropTypes.func,
-  active: React.PropTypes.bool,
-  target: React.PropTypes.string,
-  children: React.PropTypes.node.isRequired,
-};
+import React, { Component, PropTypes } from 'react';
+import { Route } from 'react-router-dom';
+
+const isModifiedEvent = (event) =>
+  !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
+
+export default class LinkContainer extends Component {
+  static contextTypes = {
+    router: PropTypes.shape({
+      history: PropTypes.shape({
+        push: PropTypes.func.isRequired,
+        replace: PropTypes.func.isRequired,
+        createHref: PropTypes.func.isRequired,
+      }).isRequired,
+    }).isRequired,
+  };
 
-const contextTypes = {
-  router: React.PropTypes.object,
-};
+  static propTypes = {
+    children: PropTypes.element.isRequired,
+    onClick: PropTypes.func,
+    target: PropTypes.string,
+    replace: PropTypes.bool,
+    to: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.object,
+    ]).isRequired,
+    exact: PropTypes.bool,
```

Comment:
Yes, I think it's worth to add default values for `exact` and `strict` props that relate to how React Router detects if route is active. However, I don't think it's worth to add default values for other non-required props just because it does not add any value and those may not be defined, so there is no need to pass them down to underlying component.

Apparently, I can set `undefined` for them in `defaultProps` but does it make any sense? Seems to just increase amount of code since these props are undefined anyway.